ecs-cli/modules/cli/compose/project/project_version.go

```diff
@@ -0,0 +1,43 @@
+package project
+
+import (
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	yaml "gopkg.in/yaml.v2"
+)
+
+func (p *ecsProject) checkComposeVersion() (string, error) {
+	var composeVersion string
+	for _, file := range p.ecsContext.ComposeFiles {
+		fileVersion, err := getFileVersion(file)
+		if err != nil {
+			return "", err
+		}
+		if len(p.ecsContext.ComposeFiles) > 0 && composeVersion != "" && composeVersion != fileVersion {
```

Comment:
Given that most customers use only one compose file, could we maybe check if `len(p.ecsContext.ComposeFiles) == 1` and return the version from `getFileVersion` early (or return an error saying there needs to be at least one ComposeFile present). Then if we do need to iterate over multiple files, that would allow us to simplify this conditional to just checking `composeVersion != "" && composeVersion != fileVersion`, I believe.

Reply:
Sorry, perhaps I wasn't clear. I was suggesting having the check for a single compose file outside the loop entirely, and only iterate if there are multiple files. E.g:
```
// For single compose file
if len(p.ecsContext.ComposeFiles) == 1 {
        composeVersion, err := getFileVersion(file)
		if err != nil {
			return "", err
		}
        return composeVersion
}

// For multiple compose files (rare, i think?)
var composeVersion string
	for _, file := range p.ecsContext.ComposeFiles {
		fileVersion, err := getFileVersion(file)
		if err != nil {
			return "", err
		}
		if composeVersion != "" && composeVersion != fileVersion {
			logrus.Errorf("Compose files must be of the same version. Found: %s and %s", composeVersion, fileVersion)
		}
		composeVersion = fileVersion
	}
	return composeVersion, nil

Reply:
But why break this up into multiple cases/blocks when one loop can handle either case? Iterating through the loop once (for one file) is just as performant as doing a standalone check. Also, if we add a check for 0 files before the loop (to detect and throw an appropriate error msg) then we can simplify the check on line 18 to `composeVersion != "" && composeVersion != fileVersion`

Reply:
Good point! I was mostly thinking of ways to avoid 3 conditions but you're right that the checking for 0 files will take care of that. Was mostly just trying to see if there was a way to write it in a more obvious way for what's happening bc my slow brain had to think about it on the first read-through, but your implementation is 👍